subprojects/platform-play/src/main/java/org/gradle/play/internal/toolchain/PlayToolChainServiceRegistry.java

```diff
@@ -38,8 +39,8 @@ public void registerProjectServices(ServiceRegistration registration) {
     }
 
     private static class ProjectScopeCompileServices {
-        PlayToolChainInternal createPlayToolChain(FileResolver fileResolver, WorkerDaemonFactory workerDaemonFactory, ConfigurationContainer configurationContainer, DependencyHandler dependencyHandler, WorkerProcessFactory workerProcessBuilderFactory) {
-            return new DefaultPlayToolChain(fileResolver, workerDaemonFactory, configurationContainer, dependencyHandler, workerProcessBuilderFactory);
+        PlayToolChainInternal createPlayToolChain(FileResolver fileResolver, WorkerDaemonFactory workerDaemonFactory, WorkerDirectoryProvider workerDirectoryProvider, ConfigurationContainer configurationContainer, DependencyHandler dependencyHandler, WorkerProcessFactory workerProcessBuilderFactory) {
+            return new DefaultPlayToolChain(fileResolver, workerDaemonFactory, workerDirectoryProvider, configurationContainer, dependencyHandler, workerProcessBuilderFactory);
```

Comment:
⭕️ It feels odd that we need to pass both a `WorkerProcessBuilderFactory`(what a name)  and a `WorkerDirectoryProvider`. Shouldn't the former know about the latter and use it internally?